src/ppx/graphics_util.cpp

```diff
@@ -639,26 +640,44 @@ Result CreateImageFromCompressedImage(
     ci.usageFlags.bits.transferSrc = true;
     ci.memoryUsage                 = grfx::MEMORY_USAGE_CPU_TO_GPU;
 
-    std::vector<MipLevel> levelSizes(mipLevelCount);
-    for (gli::texture::size_type level = 0; level < mipLevelCount; level++) {
-        auto& ls = levelSizes[level];
-
+    // Compute each mipmap level size and alignments.
+    // This step filters out levels too small to match minimal alignment.
+    std::vector<MipLevel> levelSizes;
+    for (gli::texture::size_type level = 0; level < maxMipLevelCount; level++) {
+        MipLevel ls;
         ls.width  = static_cast<uint32_t>(image.extent(level)[0]);
         ls.height = static_cast<uint32_t>(image.extent(level)[1]);
+        // Stop when mipmaps are becoming too small to respect the format alignment.
+        // The DXT* format documentation says texture sizes must be a multiple of 4.
+        // For some reason, tools like imagemagick can generate mipmaps with a size < 4.
+        // We need to ignore those.
+        if (ls.width < blockWidth || ls.height < blockWidth) {
+            break;
+        }
 
-        // bytesPerTexel is a bit misleading - for block compressed formats it's actually the
-        // number of bytes per tile. An image has to be made up of an integral number of tiles, so
-        // the actual width and height in memory are multiples of this value
-        ls.bufferWidth  = RoundUp<uint32_t>(ls.width, grfx::GetFormatDescription(format)->bytesPerTexel);
-        ls.bufferHeight = RoundUp<uint32_t>(ls.height, grfx::GetFormatDescription(format)->bytesPerTexel);
+        // If the DDS file contains textures which size is not a multiple of 4, something is wrong.
+        // Since imagemagick can create invalid mipmap levels, I'd assume it can also create invalid
+        // textures with non-multiple-of-4 sizes. Asserting to catch those.
+        PPX_ASSERT_MSG(ls.width % blockWidth == 0, "Width must me multiple of 4! Invalid DDS?");
```

Comment:
Should this just return an error when one of the dimensions isn't a multiple of 4? The assert will most likely be compiled out in release mode and it seems like a hard fail is a more obvious behavior than a possible ambiguous crash.

Reply:
Yes, good point. Changed into an error + message